frame/contracts/src/gas.rs

```diff
@@ -122,6 +125,16 @@ where
 
 	/// Absorb the remaining gas of a nested meter after we are done using it.
 	pub fn absorb_nested(&mut self, nested: Self) {
+		if self.gas_left == 0 {
+			// All of the remaining gas was inherited by the nested gas meter. When absorbing
+			// we can also safely inherit the lowest gas that the nested gas meter experienced.
+			self.gas_left_lowest = Some(nested.gas_left_lowest().min(self.gas_left_lowest()))
```

Comment:
I'm not sure this is precise enough, in case there is the parent has no yet any gas_left_lowest, then `self.gas_left_lowest()` will return 0 here. assigning 0 to self.gas_left_lowest whereas we probably want to have `nested.gas_left_lowest()`

Reply:
You are right. This is wrong. I have to think about how to fix this.

Reply:
Fixed it. I removed the `Option`, too. Simplified the code.

Reply:
yes looks good